vermouth/molecule.py

```diff
@@ -909,6 +915,59 @@ def remove_nodes_from(self, nodes):
         for node in nodes:
             self._remove_interactions_with_node(node)
 
+    def apply_link_between_residues(self, link, resids):
+        """
+        Applies a link between specific residues, if and only if
+        the link atoms incl. all attributes match at most one atom
+        in a respective link.
+
+        Parameters
+        ----------
+        link: :class:`vermouth.molecule.Link`
+            A vermouth link definition
+        resids: list
+            a list of node attributes used for link matching aside from
+            the residue ordering
+        """
+        # parked this here because importing it seems to fail apparently
+        # because some relative references are used in do_links.py
+        def _build_link_interaction_from(molecule, interaction, match):
+            atoms = tuple(match[idx] for idx in interaction.atoms)
+            parameters = [
+                param(molecule, match) if callable(param) else param
+                for param in interaction.parameters
+            ]
+            new_interaction = interaction._replace(
+                atoms=atoms,
+                parameters=parameters
+            )
+            return new_interaction
+
+        # we have to go on resid or at least one criterion otherwise
+        # the matching will be super slow, if we need to iterate
+        # over all combintions of a possible links.
+        nx.set_node_attributes(link, dict(zip(link.nodes, resids)), 'resid')
```

Comment:
Make a copy of link before doing this.
Also, since link.nodes is a dict, this depends on dict ordering. Which is bad.
Also2, if link nodes had resid attributes, those get clobbered.